### ISSUE TITLE
Can not save old dates using "text_date" cmb field

### ISSUE THREAD

**Issue by leolondon** [WebDevStudios/CMB2#744]
### Expected Behavior:

Save old dates using "text_date" cmb field

### Actual Behavior:

It seems we can't save date older than 1902 (!?)
the field is emptied once I save/update

I've tried the following:
https://github.com/WebDevStudios/CMB2/wiki/Tips-&-Tricks#change-the-year-range-for-the-date-field-types

and with this:

### CMB2 Field Registration Code:

```php
/* To increase the default year range*/

function bill_baroud02_custom_cmb2_date_format( $data ) {
	$data['defaults']['date_picker']['dateFormat'] = 'd.m.Y';
	$data['defaults']['date_picker']['yearRange'] = '0:'. ( date( 'Y' ) + 10 );


	return $data;
}
add_filter( 'cmb2_localized_data', 'bill_baroud02_custom_cmb2_date_format' );

/* Then I declare the following field within cmb2 init function */

	$cmb->add_field( array(
		'name' => esc_html__( 'Date:', 'cmb2' ),
		'desc' => esc_html__( 'Entrez ici la date du prélude', 'cmb2' ),
		'id'   => $prefix . 'date_prelude',
		'type' => 'text_date',
		'date_format' => 'd.m.Y',
	) );
```

Both methods have the same issue,
it can not save anything older than 1902
I would have understand a 1970 limit... but what happened in 1902? 
hahaha

Any thought would be great!

Thank you

Lionel

**Comment by tw2113** [comment]
I'm not managing to recreate this.

I have attached some screenshots to show the wp_postmeta row as well as the resulting metabox after saving.

![screen shot 2016-09-22 at 1 10 07 am](https://cloud.githubusercontent.com/assets/484075/18738433/56c53a4e-8061-11e6-93d5-86f05a11959f.png)
![screen shot 2016-09-22 at 1 11 12 am](https://cloud.githubusercontent.com/assets/484075/18738450/8133cc64-8061-11e6-9b74-08126d2caadd.png)

Is this a development site at the moment? Curious if some other conflict is coming in from either the current theme or some of the plugins.

**Comment by leolondon** [comment]
Exact! sorry! I've made a fresh install,
I can now set the date to 01.01.1000
the field doesn't save 21.12.128 but accept 21.12.0128

since my client has wide needs...like BC dates, or far in the future,
it would result in an endless scroll list even if I manage to fix formatting...

So I swapped to regular text for the time being, 
and just explain about date formatting in the description.

Sorry for the false alarm 
and thanks so much for your help 
and of course for awesome CMB2  !!!

Lionel
